src/Illuminate/Foundation/Exceptions/Handler.php

```diff
@@ -373,6 +373,10 @@ protected function prepareException(Throwable $e)
             $e = new HttpException(419, $e->getMessage(), $e);
         } elseif ($e instanceof SuspiciousOperationException) {
             $e = new NotFoundHttpException('Bad hostname provided.', $e);
+        } elseif ($e instanceof RecordsNotFoundException) {
+            $e = new NotFoundHttpException('Not found.', $e);
+        } elseif ($e instanceof MultipleRecordsFoundException) {
```

Comment:
This might also mean internal server error due to the database getting into a bad state?

Reply:
We could think of that. 

The use case I thought was filter constraints being passed and only one result being expected/allowed to and endpoint. Which would be a user error, not a server error.

I guess the `RecordsNotFoundException` is fine as it is, but maybe we could change the `MultipleRecordsFoundException` to be a 500 instead.

I can update the PR to whichever you find better. The goal is to have it handled gracefully as a request exception.